include/logging/log_msg2.h

```diff
@@ -120,33 +120,33 @@ union log_msg2_generic {
 };
 
 enum tracing_log_id {
```

Comment:
this enum is not being using anywhere, not sure what is the purpose of that, @nordic-krch ?

Reply:
yeah, i have started to work on having logging as tracing backend but didn't include that in logging overhaul PR. Enum was added in log2 commit and slipped through the review but it will come back.

Reply:
tracing macros were all changed, so all of this is obsolete, I would remove it now and and this later, although i think we should discuss whether this falls under a tracing backend or if it should go into the logging subsystem directly, I prefer to keep things separated and look into how to make this work with logging within the tracing subsystem.

Reply:
I removed two enumeration constants not used in the code `enum tracing_log_id` and `z_log_msg2_mode`

Reply:
Removing of the `z_log_msg2_mode` caused a CI error, removed only `tracing_log_id`